<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Analysis of DNA copy number in single cells using 
    custom genome-wide targeted DNA sequencing panels for the Mission Bio 
    Tapestri platform. Users can easily parse, manipulate, and visualize 
    datasets produced from the automated Tapestri Pipeline, with support for 
    normalization, clustering, and copy number calling. Functions are also 
    available to deconvolute multiplexed samples by genotype and parsing 
    barcoded reads from exogenous lentiviral constructs.">
<title>DNA Copy Number Analysis for Genome-Wide Tapestri Panels • karyotapR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="DNA Copy Number Analysis for Genome-Wide Tapestri Panels">
<meta property="og:description" content="Analysis of DNA copy number in single cells using 
    custom genome-wide targeted DNA sequencing panels for the Mission Bio 
    Tapestri platform. Users can easily parse, manipulate, and visualize 
    datasets produced from the automated Tapestri Pipeline, with support for 
    normalization, clustering, and copy number calling. Functions are also 
    available to deconvolute multiplexed samples by genotype and parsing 
    barcoded reads from exogenous lentiviral constructs.">
<meta property="og:image" content="http://joeymays.xyz/karyotapR/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">karyotapR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/joeymays/karyotapR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="karyotapr">karyotapR<a class="anchor" aria-label="anchor" href="#karyotapr"></a>
</h1>
</div>
<!-- badges: start -->

<p>karyotapR enables analysis of DNA copy number (aneuploidy) using custom genome-wide targeted DNA sequencing panels for the Mission Bio Tapestri system.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the current stable version of karyotapR from CRAN with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'karyotapR'</span><span class="op">)</span></span></code></pre></div>
<p>You can install the development version of karyotapR from <a href="https://github.com/joeymays/karyotapR" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"joeymays/karyotapR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="basic-usage">Basic Usage<a class="anchor" aria-label="anchor" href="#basic-usage"></a>
</h2>
<div class="section level3">
<h3 id="data-import">Data Import<a class="anchor" aria-label="anchor" href="#data-import"></a>
</h3>
<p>In practice, the dataset is imported from the <code>.h5</code> file that is generated by the Tapestri Pipeline.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/joeymays/karyotapR" class="external-link">karyotapR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://forcats.tidyverse.org/" class="external-link">forcats</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">2023</span><span class="op">)</span> <span class="co">#seed set to ensure example is reproducible </span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## NOT RUN</span></span>
<span><span class="va">example.exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/CreateTapestriExperiment.html">createTapestriExperiment</a></span><span class="op">(</span><span class="va">h5.filename</span><span class="op">)</span></span></code></pre></div>
<p>We’ll use a toy dataset for this example.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example.exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/newTapestriExperimentExample.html">newTapestriExperimentExample</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Moving gRNA probe  to `altExp` slot "grnaCounts".</span></span>
<span><span class="co">#&gt; ℹ Moving barcode probe  to `altExp` slot "barcodeCounts".</span></span>
<span><span class="co">#&gt; ℹ Moving chrY probe(s) probe_231, probe_232, probe_233, probe_234, probe_235, probe_236, probe_237, probe_238, probe_239, and probe_240 to `altExp` slot "chrYCounts".</span></span></code></pre></div>
<p>Calling the <code>TapestriExperiment</code> will print a summary of the contained data.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example.exp</span></span>
<span><span class="co">#&gt; class: TapestriExperiment </span></span>
<span><span class="co">#&gt; dim: 230 300 </span></span>
<span><span class="co">#&gt; metadata(7): sample.name pipeline.panel.name ... date.h5.created</span></span>
<span><span class="co">#&gt;   mean.reads.per.cell.per.probe</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(230): probe_1 probe_2 ... probe_229 probe_230</span></span>
<span><span class="co">#&gt; rowData names(7): probe.id chr ... total.reads median.reads</span></span>
<span><span class="co">#&gt; colnames(300): cell_1 cell_2 ... cell_299 cell_300</span></span>
<span><span class="co">#&gt; colData names(3): cell.barcode test.cluster total.reads</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: CNV</span></span>
<span><span class="co">#&gt; altExpNames(2): chrYCounts alleleFrequency</span></span>
<span><span class="co">#&gt; barcodeProbe: dummyBCprobe</span></span>
<span><span class="co">#&gt; grnaProbe: dummyGRNAprobe</span></span>
<span><span class="co">#&gt; gmmParams(0):</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="clustering">Clustering<a class="anchor" aria-label="anchor" href="#clustering"></a>
</h3>
<p>We cluster on allele frequency to partition different cell lines represented in the experiment. First, run PCA and use the knee plot to identify the PCs accounting for the most variation in the dataset.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example.exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/runPCA.html">runPCA</a></span><span class="op">(</span><span class="va">example.exp</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/PCAKneePlot.html">PCAKneePlot</a></span><span class="op">(</span><span class="va">example.exp</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-kneeplot-1.png" width="100%"></p>
<p>Run UMAP with the top PCs.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example.exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">example.exp</span>, pca.dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Running UMAP on: alleleFrequency.</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/reducedDimPlot.html">reducedDimPlot</a></span><span class="op">(</span><span class="va">example.exp</span>, dim.reduction <span class="op">=</span> <span class="st">"umap"</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-umap1-1.png" width="100%"></p>
<p>Run <code><a href="reference/runClustering.html">runClustering()</a></code> to cluster with dbscan.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example.exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/runClustering.html">runClustering</a></span><span class="op">(</span><span class="va">example.exp</span>, eps <span class="op">=</span> <span class="fl">0.9</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Finding clusters in: alleleFrequency UMAP</span></span></code></pre></div>
<p>Visualize UMAP, using “cluster” label to color.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/reducedDimPlot.html">reducedDimPlot</a></span><span class="op">(</span><span class="va">example.exp</span>, dim.reduction <span class="op">=</span> <span class="st">"umap"</span>, group.label <span class="op">=</span> <span class="st">"cluster"</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-umap2-1.png" width="100%"></p>
<p>Rename cluster labels by renaming the factor levels of “cluster”. The <code>forcats</code> package makes this easy.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">colData</span><span class="op">(</span><span class="va">example.exp</span><span class="op">)</span><span class="op">$</span><span class="va">cluster</span> <span class="op">&lt;-</span> <span class="fu">forcats</span><span class="fu">::</span><span class="fu">fct_recode</span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">example.exp</span><span class="op">)</span><span class="op">$</span><span class="va">cluster</span>, cellline1 <span class="op">=</span> <span class="st">"1"</span>, cellline2 <span class="op">=</span> <span class="st">"2"</span>, cellline3 <span class="op">=</span> <span class="st">"3"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="copy-number-calculation">Copy Number Calculation<a class="anchor" aria-label="anchor" href="#copy-number-calculation"></a>
</h2>
<p>Normalize counts and calculate copy number relative to cellline 3, which is diploid. <code>control.copy.number</code> gives the cluster label and copy number value to normalize each chromosome arm to.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example.exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/calcNormCounts.html">calcNormCounts</a></span><span class="op">(</span><span class="va">example.exp</span><span class="op">)</span></span>
<span><span class="va">control.copy.number</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/calcCopyNumber.html">generateControlCopyNumberTemplate</a></span><span class="op">(</span><span class="va">example.exp</span>, sample.feature.label <span class="op">=</span> <span class="st">"cellline3"</span>, copy.number <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">example.exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/calcCopyNumber.html">calcCopyNumber</a></span><span class="op">(</span><span class="va">example.exp</span>, control.copy.number <span class="op">=</span> <span class="va">control.copy.number</span>, sample.feature <span class="op">=</span> <span class="st">"cluster"</span><span class="op">)</span></span>
<span><span class="va">example.exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/calcSmoothCopyNumber.html">calcSmoothCopyNumber</a></span><span class="op">(</span><span class="va">example.exp</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Smoothing copy number by median...</span></span>
<span><span class="co">#&gt; ✔ Smoothing copy number by median... [1.3s]</span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
<p>Visualize copy number. Visualization reveals that cell line 1 has 1 copy of chromosome arm 1p, and cell line 2 has three copies of chromosome 7.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/assayHeatmap.html">assayHeatmap</a></span><span class="op">(</span><span class="va">example.exp</span>, assay <span class="op">=</span> <span class="st">"copyNumber"</span>, split.col.by <span class="op">=</span> <span class="st">"arm"</span>, split.row.by <span class="op">=</span> <span class="st">"cluster"</span>, annotate.row.by <span class="op">=</span> <span class="st">"cluster"</span>, color.preset <span class="op">=</span> <span class="st">"copy.number"</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-heatmaps1-1.png" width="100%"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/assayHeatmap.html">assayHeatmap</a></span><span class="op">(</span><span class="va">example.exp</span>, alt.exp <span class="op">=</span> <span class="st">"smoothedCopyNumberByArm"</span>, assay <span class="op">=</span> <span class="st">"discreteCopyNumber"</span>, split.row.by <span class="op">=</span> <span class="st">"cluster"</span>, annotate.row.by <span class="op">=</span> <span class="st">"cluster"</span>, color.preset <span class="op">=</span> <span class="st">"copy.number"</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-heatmaps1-2.png" width="100%"></p>
<p>Visualize chrY counts. Visualization reveals that cellline 2 has chr Y and is therefore male, while the other two cell lines are female.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/assayBoxPlot.html">assayBoxPlot</a></span><span class="op">(</span><span class="va">example.exp</span>, alt.exp <span class="op">=</span> <span class="st">"chrYCounts"</span>, split.features <span class="op">=</span> <span class="cn">TRUE</span>, split.x.by <span class="op">=</span> <span class="st">"cluster"</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-boxplot-1.png" width="100%"></p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/joeymays/karyotapR/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/joeymays/karyotapR/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing karyotapR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Joseph Mays <br><small class="roles"> Author, maintainer, copyright holder </small> <a href="https://orcid.org/0000-0003-4903-938X" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/joeymays/karyotapR/actions/workflows/R-CMD-check.yaml" class="external-link"><img src="https://github.com/joeymays/karyotapR/actions/workflows/R-CMD-check.yaml/badge.svg" alt="R-CMD-check"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Joseph Mays.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
