<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="countBarcodedReads() and countBarcodedReadsFromContig() match exogenous DNA barcode sequences to their associated
cell barcodes and saves them to the colData (cell barcode metadata) of TapestriExperiment.
countBarcodedReads() is a shortcut for countBarcodedReadsFromContig(), allowing the user to specify 'gRNA' or 'barcode'
to use the grnaCounts or barcodeCounts altExp slots.
The entries in the barcode.lookup table do not have to be present in the sample,
allowing users to keep one master table/file of available barcode sequences for use in all experiments.
The Rsamtools and Biostrings packages must be installed to use these functions."><title>Get read counts from barcoded reads — countBarcodedReads • karyotapR</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get read counts from barcoded reads — countBarcodedReads"><meta property="og:description" content="countBarcodedReads() and countBarcodedReadsFromContig() match exogenous DNA barcode sequences to their associated
cell barcodes and saves them to the colData (cell barcode metadata) of TapestriExperiment.
countBarcodedReads() is a shortcut for countBarcodedReadsFromContig(), allowing the user to specify 'gRNA' or 'barcode'
to use the grnaCounts or barcodeCounts altExp slots.
The entries in the barcode.lookup table do not have to be present in the sample,
allowing users to keep one master table/file of available barcode sequences for use in all experiments.
The Rsamtools and Biostrings packages must be installed to use these functions."><meta property="og:image" content="http://joeymays.xyz/karyotapR/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">karyotapR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.1.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/joeymays/karyotapR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Get read counts from barcoded reads</h1>
      <small class="dont-index">Source: <a href="https://github.com/joeymays/karyotapR/blob/HEAD/R/parseBarcodedReads.R" class="external-link"><code>R/parseBarcodedReads.R</code></a></small>
      <div class="d-none name"><code>countBarcodedReads.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>countBarcodedReads()</code> and <code>countBarcodedReadsFromContig()</code> match exogenous DNA barcode sequences to their associated
cell barcodes and saves them to the <code>colData</code> (cell barcode metadata) of <code>TapestriExperiment</code>.
<code>countBarcodedReads()</code> is a shortcut for <code>countBarcodedReadsFromContig()</code>, allowing the user to specify 'gRNA' or 'barcode'
to use the <code>grnaCounts</code> or <code>barcodeCounts</code> <code>altExp</code> slots.
The entries in the <code>barcode.lookup</code> table do not have to be present in the sample,
allowing users to keep one master table/file of available barcode sequences for use in all experiments.
The <code>Rsamtools</code> and <code>Biostrings</code> packages must be installed to use these functions.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">countBarcodedReads</span><span class="op">(</span></span>
<span>  <span class="va">TapestriExperiment</span>,</span>
<span>  <span class="va">bam.file</span>,</span>
<span>  <span class="va">barcode.lookup</span>,</span>
<span>  <span class="va">probe</span>,</span>
<span>  return.table <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  max.mismatch <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  with.indels <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">countBarcodedReadsFromContig</span><span class="op">(</span></span>
<span>  <span class="va">bam.file</span>,</span>
<span>  <span class="va">barcode.lookup</span>,</span>
<span>  <span class="va">contig</span>,</span>
<span>  cell.barcode.tag <span class="op">=</span> <span class="st">"RG"</span>,</span>
<span>  max.mismatch <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  with.indels <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>TapestriExperiment</dt>
<dd><p><code>TapestriExperiment</code> object</p></dd>


<dt>bam.file</dt>
<dd><p>File path of BAM file. <code>.bai</code> BAM index file must be in the same location (can be generated using <code><a href="https://rdrr.io/pkg/Rsamtools/man/scanBam.html" class="external-link">Rsamtools::indexBam()</a></code>).</p></dd>


<dt>barcode.lookup</dt>
<dd><p><code>data.frame</code> where the first column is the barcode identifier/name and the second column is the DNA sequence. Headers are ignored.</p></dd>


<dt>probe</dt>
<dd><p>Character, either "gRNA" or "barcode" to parse counts from <code>grnaCounts</code> or <code>barcodeCounts</code> <code>altExp</code> slots, respectively.</p></dd>


<dt>return.table</dt>
<dd><p>Logical, if <code>TRUE</code>, returns table of read counts per barcode. If <code>FALSE</code>, returns <code>TapestriExperiment.</code> Default <code>FALSE</code>.</p></dd>


<dt>max.mismatch</dt>
<dd><p>Numeric, the maximum and minimum number of mismatching letters allowed. Default 2.</p></dd>


<dt>with.indels</dt>
<dd><p>If <code>TRUE</code>, then indels are allowed. Default <code>FALSE</code>.</p></dd>


<dt>...</dt>
<dd><p>Arguments to pass on to <code>countBarcodedReadsFromContig()</code>.</p></dd>


<dt>contig</dt>
<dd><p>Character, contig or chromosome name to search for barcodes in. Can be a vector of more than one contig to expand search space.</p></dd>


<dt>cell.barcode.tag</dt>
<dd><p>Character of length 2, indicates cell barcode field in BAM, specified by Tapestri pipeline (currently "RG"). Default "RG".</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p><code>TapestriExperiment</code> with barcoded read counts added to <code>colData</code>.</p>


<p>A data.frame of read counts for each specified barcode.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="https://rdrr.io/pkg/Rsamtools/man/scanBam.html" class="external-link">Rsamtools::indexBam()</a></code></p>
<p><code><a href="https://rdrr.io/pkg/Biostrings/man/matchPattern.html" class="external-link">Biostrings::matchPattern()</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu">countBarcodedReads</span><span class="op">(</span></span></span>
<span class="r-in"><span>  <span class="va">TapestriExperiment</span>,</span></span>
<span class="r-in"><span>  <span class="va">bam.file</span>, <span class="va">barcode.lookup</span>, <span class="st">"gRNA"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu">countBarcodedReadsFromContig</span><span class="op">(</span><span class="va">bam.file</span>, <span class="va">barcode.lookup</span>, <span class="st">"virus_ref2"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Joseph Mays.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

