<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content='createTapestriExperiment() constructs a TapestriExperiment container object from data stored in the .h5 file output by the Tapestri Pipeline.
Read count matrix (probe x cell barcode) is stored in the "counts" assay slot of the top-level experiment.
Allele frequency matrix (variant x cell barcode) is stored in the "alleleFrequency" assay slot of the "alleleFrequency" altExp (alternative experiment) slot.
panel.id is an optional shortcut to set special probe identities for specific custom panels.'><title>Create TapestriExperiment object from Tapestri Pipeline output — createTapestriExperiment • karyotapR</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Create TapestriExperiment object from Tapestri Pipeline output — createTapestriExperiment"><meta property="og:description" content='createTapestriExperiment() constructs a TapestriExperiment container object from data stored in the .h5 file output by the Tapestri Pipeline.
Read count matrix (probe x cell barcode) is stored in the "counts" assay slot of the top-level experiment.
Allele frequency matrix (variant x cell barcode) is stored in the "alleleFrequency" assay slot of the "alleleFrequency" altExp (alternative experiment) slot.
panel.id is an optional shortcut to set special probe identities for specific custom panels.'><meta property="og:image" content="http://joeymays.xyz/karyotapR/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">karyotapR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/joeymays/karyotapR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Create <code>TapestriExperiment</code> object from Tapestri Pipeline output</h1>
      <small class="dont-index">Source: <a href="https://github.com/joeymays/karyotapR/blob/HEAD/R/createTapestriExperiment.R" class="external-link"><code>R/createTapestriExperiment.R</code></a></small>
      <div class="d-none name"><code>CreateTapestriExperiment.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>createTapestriExperiment()</code> constructs a <code>TapestriExperiment</code> container object from data stored in the <code>.h5</code> file output by the Tapestri Pipeline.
Read count matrix (probe x cell barcode) is stored in the "counts" <code>assay</code> slot of the top-level experiment.
Allele frequency matrix (variant x cell barcode) is stored in the "alleleFrequency" <code>assay</code> slot of the "alleleFrequency" <code>altExp</code> (alternative experiment) slot.
<code>panel.id</code> is an optional shortcut to set special probe identities for specific custom panels.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">createTapestriExperiment</span><span class="op">(</span></span>
<span>  <span class="va">h5.filename</span>,</span>
<span>  panel.id <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  get.cytobands <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  genome <span class="op">=</span> <span class="st">"hg19"</span>,</span>
<span>  move.non.genome.probes <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  filter.variants <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>h5.filename</dt>
<dd><p>File path for <code>.h5</code> file from Tapestri Pipeline output.</p></dd>


<dt>panel.id</dt>
<dd><p>Character, Tapestri panel ID, either CO261, CO293, CO610, or <code>NULL</code>. Initializes <code>barcodeProbe</code> and <code>grnaProbe</code> slots. Default <code>NULL</code>.</p></dd>


<dt>get.cytobands</dt>
<dd><p>Logical, if <code>TRUE</code> (default), retrieve and add chromosome cytobands and chromosome arms to <code>rowData</code> (probe metadata).</p></dd>


<dt>genome</dt>
<dd><p>Character, reference genome for pulling cytoband coordinates and chromosome arm labels (see <code><a href="getCytobands.html">getCytobands()</a></code>). Only "hg19" (default) is currently supported.</p></dd>


<dt>move.non.genome.probes</dt>
<dd><p>Logical, if <code>TRUE</code> (default), move counts and metadata from non-genomic probes to <code>altExp</code> slots (see <code><a href="moveNonGenomeProbes.html">moveNonGenomeProbes()</a></code>).</p></dd>


<dt>filter.variants</dt>
<dd><p>Logical, if <code>TRUE</code> (default), only stores variants that have passed Tapestri Pipeline filters.</p></dd>


<dt>verbose</dt>
<dd><p>Logical, if <code>TRUE</code> (default), metadata is output in message text.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p><code>TapestriExperiment</code> object containing data from Tapestri Pipeline output.</p>
    </div>
    <div class="section level2">
    <h2 id="panel-id-shortcuts">Panel ID Shortcuts<a class="anchor" aria-label="anchor" href="#panel-id-shortcuts"></a></h2>
    <p><code>panel.id</code> is an optional shortcut to set the <code>barcodeProbe</code> and <code>grnaProbe</code> slots in <code>TapestriExperiment</code> for specific custom Tapestri panels.</p><div class="section">
<h3 id="co-">CO261<a class="anchor" aria-label="anchor" href="#co-"></a></h3>

<ul><li><p><code>barcodeProbe</code> = "not specified"</p></li>
<li><p><code>grnaProbe</code> = "not specified"</p></li>
</ul></div>

<div class="section">
<h3 id="co--1">CO293<a class="anchor" aria-label="anchor" href="#co--1"></a></h3>

<ul><li><p><code>barcodeProbe</code> = "AMPL205334"</p></li>
<li><p><code>grnaProbe</code> = "AMPL205666"</p></li>
</ul></div>

<div class="section">
<h3 id="co--2">CO610<a class="anchor" aria-label="anchor" href="#co--2"></a></h3>

<ul><li><p><code>barcodeProbe</code> = "CO610_AMP351"</p></li>
<li><p><code>grnaProbe</code> = "CO610_AMP350"</p></li>
</ul></div>

    </div>
    <div class="section level2">
    <h2 id="automatic-operations">Automatic Operations<a class="anchor" aria-label="anchor" href="#automatic-operations"></a></h2>
    
<div class="section">
<h3 id="raw-data">Raw Data<a class="anchor" aria-label="anchor" href="#raw-data"></a></h3>


<p>Read count and allele frequency matrices are imported to their appropriate slots as described above.
<code>filter.variants == TRUE</code> (default) only loads allele frequency variants that have passed internal filters in the Tapestri Pipeline.
This greatly reduces the number of variants from tens of thousands to hundreds of likely more consequential variants,
saving RAM and reducing operation time.</p>
</div>

<div class="section">
<h3 id="metadata">Metadata<a class="anchor" aria-label="anchor" href="#metadata"></a></h3>


<p>Several metadata sets are copied or generated and then stored in the appropriate <code>TapestriExperiment</code> slot during construction.</p><ul><li><p>Probe panel metadata stored in the <code>.h5</code> file are copied to <code>rowData</code>.</p></li>
<li><p>Basic QC stats (e.g. total number of reads per probe) are added to <code>rowData.</code></p></li>
<li><p>Basic QC stats (e.g. total number of reads per cell barcode) are added to <code>colData.</code></p></li>
<li><p>Experiment-level metadata is stored in <code>metadata</code>.</p></li>
</ul></div>

    </div>
    <div class="section level2">
    <h2 id="optional-operations">Optional Operations<a class="anchor" aria-label="anchor" href="#optional-operations"></a></h2>
    <p>Two additional major operations are called by default during <code>TapestriExperiment</code> construction for convenience.
<code>get.cytobands == TRUE</code> (default) calls <code><a href="getCytobands.html">getCytobands()</a></code>, which retrieves the chromosome arm and cytoband for each probe based on stored positional data and saves them in <code>rowData</code>.
Some downstream smoothing and plotting functions may fail if chromosome arms are not present in <code>rowData</code>, so this generally should always be run.
<code>move.non.genome.probes</code> calls <code><a href="moveNonGenomeProbes.html">moveNonGenomeProbes()</a></code>, which moves probes corresponding to the specified tags to <code>altExp</code> (alternative experiment) slots in the <code>TapestriExperiment</code> object.
The exception is probes on chromosome Y; CNVs of chrY are more rare, so we move it to an <code>altExp</code> for separate analysis.
Probes corresponding to the <code>barcodeProbe</code> and <code>grnaProbe</code> slots, which are specified by the <code>panel.id</code> shortcut or manually (see <a href="slotGettersSetters.html">Custom Slot Getters and Setters</a>),
are automatically moved to <code>altExp</code> by this operation as well.
If such probes are not present, the function will only generate a warning message, so it is always safe (and recommended) to run by default.
Any remaining probes that are not targeting a human chromosome and are not specified by the shortcut tags are moved to the <code>otherProbeCounts</code> slot.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="moveNonGenomeProbes.html">moveNonGenomeProbes()</a></code>, <code><a href="getCytobands.html">getCytobands()</a></code>, which are run as part of this function by default.</p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="va">tapExperiment</span> <span class="op">&lt;-</span> <span class="fu">createTapestriExperiment</span><span class="op">(</span><span class="st">"myh5file.h5"</span>, <span class="st">"CO293"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Joseph Mays.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

