<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="karyotapR">
<title>karyotapR Basic Workflow • karyotapR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="karyotapR Basic Workflow">
<meta property="og:description" content="karyotapR">
<meta property="og:image" content="http://joeymays.xyz/karyotapR/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">karyotapR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.1.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/karyotapR.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/joeymays/karyotapR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>karyotapR Basic Workflow</h1>
            
            <h4 data-toc-skip class="date">Compiled on September 15,
2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/joeymays/karyotapR/blob/HEAD/vignettes/articles/karyotapR.Rmd" class="external-link"><code>vignettes/articles/karyotapR.Rmd</code></a></small>
      <div class="d-none name"><code>karyotapR.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/joeymays/karyotapR" class="external-link">karyotapR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://forcats.tidyverse.org/" class="external-link">forcats</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">2023</span><span class="op">)</span>  <span class="co"># seed is set to ensure tutorial is reproducible</span></span></code></pre></div>
<div class="section level3">
<h3 id="data-download">Data Download<a class="anchor" aria-label="anchor" href="#data-download"></a>
</h3>
<p>This guide uses the cell mixture experiment from the KaryoTap
publication (<a href="https://www.biorxiv.org/content/10.1101/2023.09.08.555746v1" class="external-link">Mays,
2023</a>). The Tapestri Pipeline <code>.h5</code> output file is
available on <a href="https://doi.org/10.5281/zenodo.8305841" class="external-link">Zenodo</a>
and can be downloaded by [<code><a href="https://rdrr.io/pkg/curl/man/curl_download.html" class="external-link">curl::curl_download()</a></code>] or
directly from the website.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">curl</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/curl/man/curl_download.html" class="external-link">curl_download</a></span><span class="op">(</span>url <span class="op">=</span> <span class="st">"https://zenodo.org/record/8305841/files/tapestri-experiment01-panelv1.h5?download=1"</span>,</span>
<span>    destfile <span class="op">=</span> <span class="st">"./tap-cellmixture.h5"</span>, quiet <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="basic-usage">Basic Usage<a class="anchor" aria-label="anchor" href="#basic-usage"></a>
</h2>
<div class="section level3">
<h3 id="data-import">Data Import<a class="anchor" aria-label="anchor" href="#data-import"></a>
</h3>
<p>The cell mixture dataset is imported from the <code>.h5</code> file
that is generated by the Tapestri Pipeline, which generates a new
<code>TapestriExpriment</code> object. This dataset comprises a mixture
of 5 cell lines with differing karyotypes and was processed on the
Tapestri instrument for single-cell DNA sequencing using Custom Oligo
Panel 261 (a.k.a. Panel Version 1). Setting the <code>panel.id</code>
parameter automatically assigns the correct probes to the
<code>grnaProbe</code> and <code>barcodeProbe</code> slots in the
object, which are used for special applications. Several useful
processes run automatically on import, indicated in the status messages.
For example, cytobands are automatically added to the probe metadata,
chromosome Y probes are automatically detected and moved to a specific
slot in the object (although now chrY probes exist in this panel), and
any special probes that do not target the endogenous human genome are
moved to their appropriate slots.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cellmix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CreateTapestriExperiment.html">createTapestriExperiment</a></span><span class="op">(</span><span class="st">"./tap-cellmixture.h5"</span>, panel.id <span class="op">=</span> <span class="st">"CO261"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># ── Loading Tapestri Experiment ──────────────────</span></span>
<span></span>
<span><span class="co"># • Sample Name: Teresa_s_cell_line_mix</span></span>
<span></span>
<span><span class="co"># • Pipeline Panel Name: CO261_NYU_Davoli_03102021_hg19</span></span>
<span></span>
<span><span class="co"># • Pipeline Version: 2.0.2</span></span>
<span></span>
<span><span class="co"># • Date Created: 2021-09-15</span></span>
<span></span>
<span><span class="co"># ── Metrics ──</span></span>
<span></span>
<span><span class="co"># • Number of Cells: 3555</span></span>
<span></span>
<span><span class="co"># • Number of Probes: 330</span></span>
<span></span>
<span><span class="co"># • Mean Reads per Cell per Probe: 89.22</span></span>
<span></span>
<span><span class="co"># ── Notes ── ℹ Adding cytobands from hg19.</span></span>
<span></span>
<span><span class="co"># ℹ ChrY probe ID(s) not found in TapestriExperiment object.</span></span>
<span></span>
<span><span class="co"># ℹ No non-genomic probe IDs found.</span></span></code></pre></div>
<p>Calling the object will print a summary of the contained data. The
<code>TapestriExperiment</code> class is built on top of the <a href="https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html" class="external-link"><code>SingleCellExperiment</code></a>
and <a href="https://bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html" class="external-link"><code>SummarizedExperiment</code></a>
classes, so they inherit their basic functionality and interface.
Calling <code>colData()</code> and <code>rowData()</code> will return
the metadata for the cells and probes/amplicons respectively.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cellmix</span></span>
<span><span class="co">#&gt; class: TapestriExperiment </span></span>
<span><span class="co">#&gt; dim: 330 3555 </span></span>
<span><span class="co">#&gt; metadata(7): sample.name pipeline.panel.name ... date.h5.created</span></span>
<span><span class="co">#&gt;   mean.reads.per.cell.per.probe</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(330): AMPL158802 AMPL146998 ... AMPL162086 AMPL161738</span></span>
<span><span class="co">#&gt; rowData names(8): probe.id chr ... cytoband arm</span></span>
<span><span class="co">#&gt; colnames(3555): AACAACCTAATAGTGGTT-1 AACAACCTAGTCCTAGTT-1 ...</span></span>
<span><span class="co">#&gt;   TTGTAATGCTCGTACCTT-1 TTGTATCACACTTGATCT-1</span></span>
<span><span class="co">#&gt; colData names(2): cell.barcode total.reads</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: CNV</span></span>
<span><span class="co">#&gt; altExpNames(1): alleleFrequency</span></span>
<span><span class="co">#&gt; barcodeProbe: not specified</span></span>
<span><span class="co">#&gt; grnaProbe: not specified</span></span>
<span><span class="co">#&gt; gmmParams(0):</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">colData</span><span class="op">(</span><span class="va">cellmix</span><span class="op">)</span>  <span class="co"># cell metadata</span></span>
<span><span class="co">#&gt; DataFrame with 3555 rows and 2 columns</span></span>
<span><span class="co">#&gt;                              cell.barcode total.reads</span></span>
<span><span class="co">#&gt;                               &lt;character&gt;   &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; AACAACCTAATAGTGGTT-1 AACAACCTAATAGTGGTT-1       60264</span></span>
<span><span class="co">#&gt; AACAACCTAGTCCTAGTT-1 AACAACCTAGTCCTAGTT-1       22455</span></span>
<span><span class="co">#&gt; AACAACCTATGGACGAGA-1 AACAACCTATGGACGAGA-1       51206</span></span>
<span><span class="co">#&gt; AACAACTGGCGCACTCTT-1 AACAACTGGCGCACTCTT-1       29616</span></span>
<span><span class="co">#&gt; AACAACTGGGACATAACG-1 AACAACTGGGACATAACG-1       17998</span></span>
<span><span class="co">#&gt; ...                                   ...         ...</span></span>
<span><span class="co">#&gt; TTGGTAACTCCATATCTT-1 TTGGTAACTCCATATCTT-1       44154</span></span>
<span><span class="co">#&gt; TTGTAATGCCTATAGCTC-1 TTGTAATGCCTATAGCTC-1       24469</span></span>
<span><span class="co">#&gt; TTGTAATGCGAGTGTGCC-1 TTGTAATGCGAGTGTGCC-1       45023</span></span>
<span><span class="co">#&gt; TTGTAATGCTCGTACCTT-1 TTGTAATGCTCGTACCTT-1       16216</span></span>
<span><span class="co">#&gt; TTGTATCACACTTGATCT-1 TTGTATCACACTTGATCT-1       18831</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rowData</span><span class="op">(</span><span class="va">cellmix</span><span class="op">)</span>  <span class="co"># probe metadata</span></span>
<span><span class="co">#&gt; DataFrame with 330 rows and 8 columns</span></span>
<span><span class="co">#&gt;               probe.id      chr start.pos   end.pos total.reads median.reads</span></span>
<span><span class="co">#&gt;            &lt;character&gt; &lt;factor&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;    &lt;integer&gt;</span></span>
<span><span class="co">#&gt; AMPL158802  AMPL158802        1   1479191   1479385      202832           49</span></span>
<span><span class="co">#&gt; AMPL146998  AMPL146998        1   6196653   6196900      112408           24</span></span>
<span><span class="co">#&gt; AMPL158817  AMPL158817        1  11832076  11832330       99472           22</span></span>
<span><span class="co">#&gt; AMPL158827  AMPL158827        1  17087135  17087388     1162986          304</span></span>
<span><span class="co">#&gt; AMPL158845  AMPL158845        1  24412922  24413191        1008            0</span></span>
<span><span class="co">#&gt; ...                ...      ...       ...       ...         ...          ...</span></span>
<span><span class="co">#&gt; AMPL161732  AMPL161732        X 128880366 128880635      568729          133</span></span>
<span><span class="co">#&gt; AMPL161734  AMPL161734        X 130415574 130415843      231144           49</span></span>
<span><span class="co">#&gt; AMPL161735  AMPL161735        X 135429800 135430069      156462           33</span></span>
<span><span class="co">#&gt; AMPL162086  AMPL162086        X 140993717 140993974      195494           47</span></span>
<span><span class="co">#&gt; AMPL161738  AMPL161738        X 142794979 142795248      400235           93</span></span>
<span><span class="co">#&gt;               cytoband      arm</span></span>
<span><span class="co">#&gt;            &lt;character&gt; &lt;factor&gt;</span></span>
<span><span class="co">#&gt; AMPL158802      p36.33       1p</span></span>
<span><span class="co">#&gt; AMPL146998      p36.31       1p</span></span>
<span><span class="co">#&gt; AMPL158817      p36.22       1p</span></span>
<span><span class="co">#&gt; AMPL158827      p36.13       1p</span></span>
<span><span class="co">#&gt; AMPL158845      p36.11       1p</span></span>
<span><span class="co">#&gt; ...                ...      ...</span></span>
<span><span class="co">#&gt; AMPL161732       q26.1       Xq</span></span>
<span><span class="co">#&gt; AMPL161734       q26.2       Xq</span></span>
<span><span class="co">#&gt; AMPL161735       q26.3       Xq</span></span>
<span><span class="co">#&gt; AMPL162086       q27.2       Xq</span></span>
<span><span class="co">#&gt; AMPL161738       q27.3       Xq</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="allele-frequency-clustering">Allele Frequency Clustering<a class="anchor" aria-label="anchor" href="#allele-frequency-clustering"></a>
</h3>
<p>We cluster on allele frequency to partition different cell lines
represented in the experiment. First, we run Principal Components
Analysis (PCA) and use the knee plot to identify the principal
components (PCs) accounting for the most variation in the dataset.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cellmix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runPCA.html">runPCA</a></span><span class="op">(</span><span class="va">cellmix</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/PCAKneePlot.html">PCAKneePlot</a></span><span class="op">(</span><span class="va">cellmix</span><span class="op">)</span></span></code></pre></div>
<p><img src="karyotapR_files/figure-html/pca-1.png" width="700"></p>
<p>Next, we run UMAP with the top PCs to embed them into two dimensions
and plot the result.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cellmix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">cellmix</span>, pca.dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Running UMAP on: <span style="font-style: italic;">alleleFrequency</span>.</span></span>
<span><span class="fu"><a href="../reference/reducedDimPlot.html">reducedDimPlot</a></span><span class="op">(</span><span class="va">cellmix</span>, dim.reduction <span class="op">=</span> <span class="st">"umap"</span><span class="op">)</span></span></code></pre></div>
<p><img src="karyotapR_files/figure-html/umap1-1.png" width="700"></p>
<p>Next, we partition the data into clusters using the dbscan method.
The <code>eps</code> parameter can be used to adjust the granularity of
the clustering. We can then update the UMAP plot with the clusters.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cellmix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runClustering.html">runClustering</a></span><span class="op">(</span><span class="va">cellmix</span>, eps <span class="op">=</span> <span class="fl">0.9</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Finding clusters in: <span style="font-style: italic;">alleleFrequency</span> <span style="font-style: italic;">UMAP</span></span></span>
<span><span class="fu"><a href="../reference/reducedDimPlot.html">reducedDimPlot</a></span><span class="op">(</span><span class="va">cellmix</span>, dim.reduction <span class="op">=</span> <span class="st">"umap"</span>, group.label <span class="op">=</span> <span class="st">"cluster"</span><span class="op">)</span></span></code></pre></div>
<p><img src="karyotapR_files/figure-html/clustering-1.png" width="700"></p>
<p>As expected, we have 5 major clusters corresponding to the 5 cell
lines in the sequencing run, with the smaller clusters representing
doublets (i.e., two cells in sequenced together in one droplet). We can
subset the doublets out by pulling the cell barcodes corresponding to
clusters 1-5 (clusters are ordered and named by descending size) and
using those to subset the “columns” of the object into a new object.
This is done here using a logical vector, but the cell barcodes can be
passed in as a character vector of barcodes as well.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cellmix.subset</span> <span class="op">&lt;-</span> <span class="va">cellmix</span><span class="op">[</span>, <span class="fu">colData</span><span class="op">(</span><span class="va">cellmix</span><span class="op">)</span><span class="op">$</span><span class="va">cluster</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<p>We’ll rename the cluster labels by renaming the factor levels of the
“cluster” column in the <code>colData</code> slot, print an updated
plot, and count the number of cells in each cluster.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">colData</span><span class="op">(</span><span class="va">cellmix.subset</span><span class="op">)</span><span class="op">$</span><span class="va">cluster</span> <span class="op">&lt;-</span> <span class="fu">forcats</span><span class="fu">::</span><span class="fu"><a href="https://forcats.tidyverse.org/reference/fct_recode.html" class="external-link">fct_recode</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">cellmix.subset</span><span class="op">)</span><span class="op">$</span><span class="va">cluster</span>,</span>
<span>    cellline1 <span class="op">=</span> <span class="st">"1"</span>, cellline2 <span class="op">=</span> <span class="st">"2"</span>, cellline3 <span class="op">=</span> <span class="st">"3"</span>, cellline4 <span class="op">=</span> <span class="st">"4"</span>, cellline5 <span class="op">=</span> <span class="st">"5"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/reducedDimPlot.html">reducedDimPlot</a></span><span class="op">(</span><span class="va">cellmix.subset</span>, dim.reduction <span class="op">=</span> <span class="st">"umap"</span>, group.label <span class="op">=</span> <span class="st">"cluster"</span><span class="op">)</span></span></code></pre></div>
<p><img src="karyotapR_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">forcats</span><span class="fu">::</span><span class="fu"><a href="https://forcats.tidyverse.org/reference/fct_count.html" class="external-link">fct_count</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">cellmix.subset</span><span class="op">)</span><span class="op">$</span><span class="va">cluster</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 12 × 2</span></span></span>
<span><span class="co">#&gt;    f             n</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> cellline1   987</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> cellline2   591</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> cellline3   575</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> cellline4   450</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> cellline5   398</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> 6             0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> 7             0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> 8             0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> 9             0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> 10            0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> 11            0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> 12            0</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="copy-number-calling">Copy Number Calling<a class="anchor" aria-label="anchor" href="#copy-number-calling"></a>
</h3>
<p>The KaryoTap method works best with a reference population where the
copy number for each chromosome arm is known. Here we used RPE1 cells
which are diploid (2 copies) except for a third copy of the chromosome
10q arm. We know from the KaryoTap preprint that “cellline2” corresponds
to the RPE1 cells.</p>
<p>Her we normalize the read counts in the object and calculate a copy
number score relative to cellline 2. <code>control.copy.number</code>
gives the cluster label and copy number value to normalize each
chromosome arm to. <code><a href="../reference/calcCopyNumber.html">generateControlCopyNumberTemplate()</a></code>
creates a dataframe that is used to indicate the copy number of the
reference population. The entry for chr10q has to be changed to 3.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cellmix.subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calcNormCounts.html">calcNormCounts</a></span><span class="op">(</span><span class="va">cellmix.subset</span><span class="op">)</span></span>
<span><span class="va">control.copy.number</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calcCopyNumber.html">generateControlCopyNumberTemplate</a></span><span class="op">(</span><span class="va">cellmix.subset</span>, sample.feature.label <span class="op">=</span> <span class="st">"cellline3"</span>,</span>
<span>    copy.number <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">control.copy.number</span><span class="op">[</span><span class="st">"10q"</span>, <span class="st">"copy.number"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">3</span></span></code></pre></div>
<p>The <code>calcCopyNumber</code> function will throw an error if the
median normalized counts for a probe in the reference population is
zero, which would otherwise result in a division-by-zero calculation
error. These probes need to be removed before moving forward.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/try.html" class="external-link">try</a></span><span class="op">(</span><span class="va">cellmix.subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calcCopyNumber.html">calcCopyNumber</a></span><span class="op">(</span><span class="va">cellmix.subset</span>, control.copy.number <span class="op">=</span> <span class="va">control.copy.number</span>,</span>
<span>    sample.feature <span class="op">=</span> <span class="st">"cluster"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Error in calcCopyNumber(cellmix.subset, control.copy.number = control.copy.number,  : </span></span>
<span><span class="co">#&gt;   AMPL158845, AMPL147043, AMPL147154, AMPL159975, AMPL147293, AMPL113086,</span></span>
<span><span class="co">#&gt; AMPL147323, AMPL158390, and AMPL158655 control cell median equal to 0. This</span></span>
<span><span class="co">#&gt; will cause a division-by-zero error. Filter out prior to proceeding.</span></span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">probes.to.remove</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AMPL158845"</span>, <span class="st">"AMPL147043"</span>, <span class="st">"AMPL147154"</span>, <span class="st">"AMPL159975"</span>, <span class="st">"AMPL147293"</span>,</span>
<span>    <span class="st">"AMPL113086"</span>, <span class="st">"AMPL147323"</span>, <span class="st">"AMPL158390"</span>, <span class="st">"AMPL158655"</span><span class="op">)</span></span>
<span><span class="va">cellmix.subset</span> <span class="op">&lt;-</span> <span class="va">cellmix.subset</span><span class="op">[</span><span class="op">!</span><span class="fu">rowData</span><span class="op">(</span><span class="va">cellmix.subset</span><span class="op">)</span><span class="op">$</span><span class="va">probe.id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">probes.to.remove</span>,</span>
<span>    <span class="op">]</span></span>
<span></span>
<span><span class="va">cellmix.subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calcCopyNumber.html">calcCopyNumber</a></span><span class="op">(</span><span class="va">cellmix.subset</span>, control.copy.number <span class="op">=</span> <span class="va">control.copy.number</span>,</span>
<span>    sample.feature <span class="op">=</span> <span class="st">"cluster"</span><span class="op">)</span></span></code></pre></div>
<p>The <code><a href="../reference/calcNormCounts.html">calcNormCounts()</a></code> and <code><a href="../reference/calcCopyNumber.html">calcCopyNumber()</a></code>
functions take the count matrix in the main <code>assay</code> slot of
of the <code>TapestriExperiment</code>, perform their operation, and
save the result to new <code>assay</code> slots, which can be accessed
using <code>assay()</code> or listed using <code>assays()</code>. Assays
in the <code>SingleCellExperiment</code> sense are sets of measurements
for the same set of samples (columns) and features (rows). For the
<code>counts</code>, <code>normcounts</code> and <code>copyNumber</code>
assays, the features (probes) and samples (cell barcodes) are the
same.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">assays</span><span class="op">(</span><span class="va">cellmix.subset</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of length 3</span></span>
<span><span class="co">#&gt; names(3): counts normcounts copyNumber</span></span></code></pre></div>
<p><code><a href="../reference/calcSmoothCopyNumber.html">calcSmoothCopyNumber()</a></code> produces one smoothed copy number
score for each chromosome and cell. Since the features here are
chromosomes, no longer probes, the values get saved to an
<code>altExp</code> (alternate experiment) slot, which allows for
measurements from the same samples (cell barcodes), with a different
feature set than the top-level experiment in the
<code>TapestriExperiment</code> object (i.e. probes
vs. chromosomes).</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cellmix.subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calcSmoothCopyNumber.html">calcSmoothCopyNumber</a></span><span class="op">(</span><span class="va">cellmix.subset</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Smoothing copy number by median...</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Smoothing copy number by median... <span style="color: #B2B2B2;">[9.8s]</span></span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
<p>Visualization of the copy number scores reveals the heterogeneity.
Here we’re showing the copy number scores by probe, and the smoothed
copy number scores by whole chromosome and by chromosome arm. See the
documentation for <code><a href="../reference/assayHeatmap.html">assayHeatmap()</a></code> for details on
customization.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/assayHeatmap.html">assayHeatmap</a></span><span class="op">(</span><span class="va">cellmix.subset</span>, assay <span class="op">=</span> <span class="st">"copyNumber"</span>, split.col.by <span class="op">=</span> <span class="st">"arm"</span>, split.row.by <span class="op">=</span> <span class="st">"cluster"</span>,</span>
<span>    annotate.row.by <span class="op">=</span> <span class="st">"cluster"</span>, color.preset <span class="op">=</span> <span class="st">"copy.number"</span><span class="op">)</span></span></code></pre></div>
<p><img src="karyotapR_files/figure-html/heatmaps%201-1.png" width="700"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/assayHeatmap.html">assayHeatmap</a></span><span class="op">(</span><span class="va">cellmix.subset</span>, alt.exp <span class="op">=</span> <span class="st">"smoothedCopyNumberByChr"</span>, assay <span class="op">=</span> <span class="st">"smoothedCopyNumber"</span>,</span>
<span>    split.row.by <span class="op">=</span> <span class="st">"cluster"</span>, annotate.row.by <span class="op">=</span> <span class="st">"cluster"</span>, color.preset <span class="op">=</span> <span class="st">"copy.number"</span><span class="op">)</span></span></code></pre></div>
<p><img src="karyotapR_files/figure-html/heatmaps%201-2.png" width="700"></p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/assayHeatmap.html">assayHeatmap</a></span><span class="op">(</span><span class="va">cellmix.subset</span>, alt.exp <span class="op">=</span> <span class="st">"smoothedCopyNumberByArm"</span>, assay <span class="op">=</span> <span class="st">"smoothedCopyNumber"</span>,</span>
<span>    split.row.by <span class="op">=</span> <span class="st">"cluster"</span>, annotate.row.by <span class="op">=</span> <span class="st">"cluster"</span>, color.preset <span class="op">=</span> <span class="st">"copy.number"</span><span class="op">)</span></span></code></pre></div>
<p><img src="karyotapR_files/figure-html/heatmaps%201-3.png" width="700"></p>
<p>Finally, a integer copy number value for each chromosome in each cell
can be called using a Gaussian Mixture Model (GMM) framework.
<code><a href="../reference/calcGMMCopyNumber.html">calcGMMCopyNumber()</a></code> takes a vector of cell barcodes for the
reference sample and a template data frame from
<code><a href="../reference/calcCopyNumber.html">generateControlCopyNumberTemplate()</a></code> indicating the copy
number of each chromosome arm in the reference. Here we can use the same
<code>control.copy.number</code> template that was generated earlier.
Here we are specifying a model that has copy number = {1,2,3,4}. The
results of this are saved as new <code>assays</code> in the
“smoothedCopyNumber” <code>altExp</code> slots for chromosomes and
chromosome arms.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">reference.bcs</span> <span class="op">&lt;-</span> <span class="fu">colData</span><span class="op">(</span><span class="va">cellmix.subset</span><span class="op">)</span><span class="op">$</span><span class="va">cell.barcode</span><span class="op">[</span><span class="fu">colData</span><span class="op">(</span><span class="va">cellmix.subset</span><span class="op">)</span><span class="op">$</span><span class="va">cluster</span> <span class="op">==</span></span>
<span>    <span class="st">"cellline2"</span><span class="op">]</span></span>
<span><span class="va">cellmix.subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calcGMMCopyNumber.html">calcGMMCopyNumber</a></span><span class="op">(</span><span class="va">cellmix.subset</span>, cell.barcodes <span class="op">=</span> <span class="va">reference.bcs</span>,</span>
<span>    control.copy.number <span class="op">=</span> <span class="va">control.copy.number</span>, model.components <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Calculating GMMs using 591 input cells.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Generating probe values for 500 simulated cells...</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Generating probe values for 500 simulated cells... <span style="color: #B2B2B2;">[2.3s]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Fitting Gaussian distributions to simulated cells...</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Fitting Gaussian distributions to simulated cells... <span style="color: #B2B2B2;">[12.9s]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Calculating posterior probabilities...</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Calculating posterior probabilities... <span style="color: #B2B2B2;">[662ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Calling copy number from posterior probabilities...</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Calling copy number from posterior probabilities... <span style="color: #B2B2B2;">[213ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Saving whole chromosome copy number calls to altExp: smoothedCopyNumberByChr,</span></span>
<span><span class="co">#&gt;   assay: gmmCopyNumber...</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Saving chromosome arm copy number calls to altExp: smoothedCopyNumberByArm,</span></span>
<span><span class="co">#&gt;   assay: gmmCopyNumber...</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Saving GMM models and metadata to `gmmParams` slot...</span></span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/assayHeatmap.html">assayHeatmap</a></span><span class="op">(</span><span class="va">cellmix.subset</span>, alt.exp <span class="op">=</span> <span class="st">"smoothedCopyNumberByChr"</span>, assay <span class="op">=</span> <span class="st">"gmmCopyNumber"</span>,</span>
<span>    split.row.by <span class="op">=</span> <span class="st">"cluster"</span>, annotate.row.by <span class="op">=</span> <span class="st">"cluster"</span>, color.preset <span class="op">=</span> <span class="st">"copy.number"</span><span class="op">)</span></span></code></pre></div>
<p><img src="karyotapR_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/assayHeatmap.html">assayHeatmap</a></span><span class="op">(</span><span class="va">cellmix.subset</span>, alt.exp <span class="op">=</span> <span class="st">"smoothedCopyNumberByArm"</span>, assay <span class="op">=</span> <span class="st">"gmmCopyNumber"</span>,</span>
<span>    split.row.by <span class="op">=</span> <span class="st">"cluster"</span>, annotate.row.by <span class="op">=</span> <span class="st">"cluster"</span>, color.preset <span class="op">=</span> <span class="st">"copy.number"</span><span class="op">)</span></span></code></pre></div>
<p><img src="karyotapR_files/figure-html/unnamed-chunk-12-2.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sessioninfo</span><span class="fu">::</span><span class="fu"><a href="https://r-lib.github.io/sessioninfo/reference/session_info.html" class="external-link">session_info</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">#&gt;  os       Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">#&gt;  system   x86_64, linux-gnu</span></span>
<span><span class="co">#&gt;  ui       X11</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  C.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    C.UTF-8</span></span>
<span><span class="co">#&gt;  tz       UTC</span></span>
<span><span class="co">#&gt;  date     2023-09-15</span></span>
<span><span class="co">#&gt;  pandoc   2.19.2 @ /usr/bin/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package             </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version   </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;  abind                  1.4-5      <span style="color: #555555;">2016-07-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  askpass                1.2.0      <span style="color: #555555;">2023-09-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  Biobase              * 2.60.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocGenerics         * 0.46.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  bitops                 1.0-7      <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  bslib                  0.5.1      <span style="color: #555555;">2023-08-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  cachem                 1.0.8      <span style="color: #555555;">2023-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  circlize               0.4.15     <span style="color: #555555;">2022-05-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  cli                    3.6.1      <span style="color: #555555;">2023-03-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  clue                   0.3-64     <span style="color: #555555;">2023-01-31</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  cluster                2.1.4      <span style="color: #555555;">2022-08-22</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  codetools              0.2-19     <span style="color: #555555;">2023-02-01</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  colorspace             2.1-0      <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  ComplexHeatmap         2.16.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  crayon                 1.5.2      <span style="color: #555555;">2022-09-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  dbscan                 1.1-11     <span style="color: #555555;">2022-10-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  DelayedArray           0.26.7     <span style="color: #555555;">2023-07-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  desc                   1.4.2      <span style="color: #555555;">2022-09-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  digest                 0.6.33     <span style="color: #555555;">2023-07-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  doParallel             1.0.17     <span style="color: #555555;">2022-02-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  dplyr                  1.1.3      <span style="color: #555555;">2023-09-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  evaluate               0.21       <span style="color: #555555;">2023-05-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  fansi                  1.0.4      <span style="color: #555555;">2023-01-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  farver                 2.1.1      <span style="color: #555555;">2022-07-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  fastmap                1.1.1      <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  fitdistrplus           1.1-11     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  forcats              * 1.0.0      <span style="color: #555555;">2023-01-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  foreach                1.5.2      <span style="color: #555555;">2022-02-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  formatR                1.14       <span style="color: #555555;">2023-01-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  fs                     1.6.3      <span style="color: #555555;">2023-07-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  generics               0.1.3      <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDb         * 1.36.3     <span style="color: #555555;">2023-09-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDbData       1.2.10     <span style="color: #555555;">2023-08-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomicRanges        * 1.52.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GetoptLong             1.0.5      <span style="color: #555555;">2020-12-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  ggplot2                3.4.3      <span style="color: #555555;">2023-08-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  GlobalOptions          0.1.2      <span style="color: #555555;">2020-06-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  glue                   1.6.2      <span style="color: #555555;">2022-02-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  gtable                 0.3.4      <span style="color: #555555;">2023-08-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  htmltools              0.5.6      <span style="color: #555555;">2023-08-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  IRanges              * 2.34.1     <span style="color: #555555;">2023-06-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  iterators              1.0.14     <span style="color: #555555;">2022-02-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  jquerylib              0.1.4      <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  jsonlite               1.8.7      <span style="color: #555555;">2023-06-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  karyotapR            * <span style="color: #BB00BB; font-weight: bold;">1.0.1.9000</span> <span style="color: #555555;">2023-09-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">local</span></span></span>
<span><span class="co">#&gt;  knitr                  1.44       <span style="color: #555555;">2023-09-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  labeling               0.4.3      <span style="color: #555555;">2023-08-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  lattice                0.21-8     <span style="color: #555555;">2023-04-05</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  lifecycle              1.0.3      <span style="color: #555555;">2022-10-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  magrittr               2.0.3      <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  MASS                   7.3-60     <span style="color: #555555;">2023-05-04</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  Matrix                 1.5-4.1    <span style="color: #555555;">2023-05-18</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  MatrixGenerics       * 1.12.3     <span style="color: #555555;">2023-07-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  matrixStats          * 1.0.0      <span style="color: #555555;">2023-06-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  memoise                2.0.1      <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  munsell                0.5.0      <span style="color: #555555;">2018-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  openssl                2.1.0      <span style="color: #555555;">2023-07-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  pillar                 1.9.0      <span style="color: #555555;">2023-03-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  pkgconfig              2.0.3      <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  pkgdown                2.0.7      <span style="color: #555555;">2022-12-14</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">any (@2.0.7)</span></span></span>
<span><span class="co">#&gt;  png                    0.1-8      <span style="color: #555555;">2022-11-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  purrr                  1.0.2      <span style="color: #555555;">2023-08-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  R6                     2.5.1      <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  ragg                   1.2.5      <span style="color: #555555;">2023-01-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  RColorBrewer           1.1-3      <span style="color: #555555;">2022-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  Rcpp                   1.0.11     <span style="color: #555555;">2023-07-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  RCurl                  1.98-1.12  <span style="color: #555555;">2023-03-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  reticulate             1.32.0     <span style="color: #555555;">2023-09-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  rjson                  0.2.21     <span style="color: #555555;">2022-01-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  rlang                  1.1.1      <span style="color: #555555;">2023-04-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  rmarkdown              2.24       <span style="color: #555555;">2023-08-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  rprojroot              2.0.3      <span style="color: #555555;">2022-04-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  RSpectra               0.16-1     <span style="color: #555555;">2022-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  S4Arrays               1.0.6      <span style="color: #555555;">2023-08-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  S4Vectors            * 0.38.1     <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  sass                   0.4.7      <span style="color: #555555;">2023-07-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  scales                 1.2.1      <span style="color: #555555;">2022-08-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  sessioninfo            1.2.2      <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">any (@1.2.2)</span></span></span>
<span><span class="co">#&gt;  shape                  1.4.6      <span style="color: #555555;">2021-05-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  SingleCellExperiment * 1.22.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  stringi                1.7.12     <span style="color: #555555;">2023-01-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  stringr                1.5.0      <span style="color: #555555;">2022-12-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  SummarizedExperiment * 1.30.2     <span style="color: #555555;">2023-06-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  survival               3.5-5      <span style="color: #555555;">2023-03-12</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  systemfonts            1.0.4      <span style="color: #555555;">2022-02-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  textshaping            0.3.6      <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  tibble                 3.2.1      <span style="color: #555555;">2023-03-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  tidyr                  1.3.0      <span style="color: #555555;">2023-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  tidyselect             1.2.0      <span style="color: #555555;">2022-10-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  umap                   0.2.10.0   <span style="color: #555555;">2023-02-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  utf8                   1.2.3      <span style="color: #555555;">2023-01-31</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  vctrs                  0.6.3      <span style="color: #555555;">2023-06-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  viridisLite            0.4.2      <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  withr                  2.5.0      <span style="color: #555555;">2022-03-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  xfun                   0.40       <span style="color: #555555;">2023-08-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  XVector                0.40.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  yaml                   2.3.7      <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.1)</span></span></span>
<span><span class="co">#&gt;  zlibbioc               1.46.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] /home/runner/work/_temp/Library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [2] /opt/R/4.3.1/lib/R/site-library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [3] /opt/R/4.3.1/lib/R/library</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Joseph Mays.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
