<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="karyotapR">
<title>Navigating the TapestriExperiment Class • karyotapR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Navigating the TapestriExperiment Class">
<meta property="og:description" content="karyotapR">
<meta property="og:image" content="http://joeymays.xyz/karyotapR/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">karyotapR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.1.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/karyotapR.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/GMMs-for-Copy-Number-Calling.html">GMMs for Copy Number Calling</a>
    <a class="dropdown-item" href="../articles/TapExp-navigation.html">Navigating the TapestriExperiment Class</a>
    <a class="dropdown-item" href="../articles/barcoding.html">Barcoding and gRNA Probes</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/joeymays/karyotapR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Navigating the TapestriExperiment Class</h1>
            
            <h4 data-toc-skip class="date">Compiled on September 18,
2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/joeymays/karyotapR/blob/HEAD/vignettes/articles/TapExp-navigation.Rmd" class="external-link"><code>vignettes/articles/TapExp-navigation.Rmd</code></a></small>
      <div class="d-none name"><code>TapExp-navigation.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The <code>TapestriExperiment</code> object class is the container
that holds all data and metadata related to a KaryoTap experiment. The
<code>TapestriExperiment</code> class is built on top of the<a href="https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html" class="external-link"><code>SingleCellExperiment</code></a>
and <a href="https://bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html" class="external-link"><code>SummarizedExperiment</code></a>
classes, and they inherit their basic functionality and interface. More
information can be found in their respective documentations. This
tutorial will cover the basics to the structure and navigation of the
<code>TapestriExperiment</code> class.</p>
</div>
<div class="section level2">
<h2 id="tapestriexperiment">TapestriExperiment<a class="anchor" aria-label="anchor" href="#tapestriexperiment"></a>
</h2>
<p>We’ll use the cell mixture experiment from the KaryoTap publication
as an example.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/joeymays/karyotapR" class="external-link">karyotapR</a></span><span class="op">)</span></span></code></pre></div>
<p>Calling the object will print a summary of the contained data.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cellmix</span></span>
<span><span class="co">#&gt; class: TapestriExperiment </span></span>
<span><span class="co">#&gt; dim: 317 2987 </span></span>
<span><span class="co">#&gt; metadata(7): sample.name pipeline.panel.name ... date.h5.created mean.reads.per.cell.per.probe</span></span>
<span><span class="co">#&gt; assays(3): counts normcounts copyNumber</span></span>
<span><span class="co">#&gt; rownames(317): AMPL158802 AMPL146998 ... AMPL162086 AMPL161738</span></span>
<span><span class="co">#&gt; rowData names(9): probe.id chr ... arm norm.count.sd</span></span>
<span><span class="co">#&gt; colnames(2987): AACAACCTAATAGTGGTT-1 AACAACCTAGTCCTAGTT-1 ... TTGTAATGCTCGTACCTT-1</span></span>
<span><span class="co">#&gt;   TTGTATCACACTTGATCT-1</span></span>
<span><span class="co">#&gt; colData names(3): cell.barcode total.reads cluster</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: CNV</span></span>
<span><span class="co">#&gt; altExpNames(3): alleleFrequency smoothedCopyNumberByChr smoothedCopyNumberByArm</span></span>
<span><span class="co">#&gt; barcodeProbe: not specified</span></span>
<span><span class="co">#&gt; grnaProbe: not specified</span></span>
<span><span class="co">#&gt; gmmParams(2): chr arm</span></span></code></pre></div>
<p>The centerpiece of the object is the <code>assay</code> slot. This is
organized as a matrix, and calling <code><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim()</a></code> on the object will
accordingly return the dimensions. For <code>TapestriExperiment</code>,
the rows represent probes (i.e., features) and the columns represent
cells (i.e., samples). Calling <code>colData()</code> and
<code>rowData()</code> will return the metadata for the cells and probes
respectively.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">cellmix</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1]  317 2987</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">colData</span><span class="op">(</span><span class="va">cellmix</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 2987 rows and 3 columns</span></span>
<span><span class="co">#&gt;                              cell.barcode total.reads  cluster</span></span>
<span><span class="co">#&gt;                               &lt;character&gt;   &lt;numeric&gt; &lt;factor&gt;</span></span>
<span><span class="co">#&gt; AACAACCTAATAGTGGTT-1 AACAACCTAATAGTGGTT-1       60264    LS513</span></span>
<span><span class="co">#&gt; AACAACCTAGTCCTAGTT-1 AACAACCTAGTCCTAGTT-1       22455    LS513</span></span>
<span><span class="co">#&gt; AACAACCTATGGACGAGA-1 AACAACCTATGGACGAGA-1       51206    LS513</span></span>
<span><span class="co">#&gt; AACAACTGGGACATAACG-1 AACAACTGGGACATAACG-1       17998    CL11 </span></span>
<span><span class="co">#&gt; AACAACTGGGGAACCTAG-1 AACAACTGGGGAACCTAG-1       14202    LoVo </span></span>
<span><span class="co">#&gt; ...                                   ...         ...      ...</span></span>
<span><span class="co">#&gt; TTGGAGAACTGGTAGCAG-1 TTGGAGAACTGGTAGCAG-1       26435     RPE1</span></span>
<span><span class="co">#&gt; TTGGTAACTCCATATCTT-1 TTGGTAACTCCATATCTT-1       44154     SW48</span></span>
<span><span class="co">#&gt; TTGTAATGCCTATAGCTC-1 TTGTAATGCCTATAGCTC-1       24469     RPE1</span></span>
<span><span class="co">#&gt; TTGTAATGCTCGTACCTT-1 TTGTAATGCTCGTACCTT-1       16216     SW48</span></span>
<span><span class="co">#&gt; TTGTATCACACTTGATCT-1 TTGTATCACACTTGATCT-1       18831     SW48</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rowData</span><span class="op">(</span><span class="va">cellmix</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 317 rows and 9 columns</span></span>
<span><span class="co">#&gt;               probe.id      chr start.pos   end.pos total.reads median.reads    cytoband      arm norm.count.sd</span></span>
<span><span class="co">#&gt;            &lt;character&gt; &lt;factor&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;    &lt;integer&gt; &lt;character&gt; &lt;factor&gt;     &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; AMPL158802  AMPL158802        1   1479191   1479385      202832           49      p36.33    chr1p      1.365966</span></span>
<span><span class="co">#&gt; AMPL146998  AMPL146998        1   6196653   6196900      112408           24      p36.31    chr1p      1.264583</span></span>
<span><span class="co">#&gt; AMPL158817  AMPL158817        1  11832076  11832330       99472           22      p36.22    chr1p      1.093818</span></span>
<span><span class="co">#&gt; AMPL158827  AMPL158827        1  17087135  17087388     1162986          304      p36.13    chr1p      0.665911</span></span>
<span><span class="co">#&gt; AMPL147006  AMPL147006        1  34285091  34285360      248023           53       p35.1    chr1p      1.181844</span></span>
<span><span class="co">#&gt; ...                ...      ...       ...       ...         ...          ...         ...      ...           ...</span></span>
<span><span class="co">#&gt; AMPL161732  AMPL161732        X 128880366 128880635      568729          133       q26.1    chrXq       1.14576</span></span>
<span><span class="co">#&gt; AMPL161734  AMPL161734        X 130415574 130415843      231144           49       q26.2    chrXq       1.34813</span></span>
<span><span class="co">#&gt; AMPL161735  AMPL161735        X 135429800 135430069      156462           33       q26.3    chrXq       1.80859</span></span>
<span><span class="co">#&gt; AMPL162086  AMPL162086        X 140993717 140993974      195494           47       q27.2    chrXq       1.24238</span></span>
<span><span class="co">#&gt; AMPL161738  AMPL161738        X 142794979 142795248      400235           93       q27.3    chrXq       1.33365</span></span></code></pre></div>
<p>Metadata columns can be added to either <code>rowData</code> or
<code>colData</code> by assignment</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rowData</span><span class="op">(</span><span class="va">cellmix</span><span class="op">)</span><span class="op">$</span><span class="va">example.name</span> <span class="op">&lt;-</span> <span class="st">"example.value"</span></span></code></pre></div>
<p>The entire object can be subset using bracket notation, either by
index or with a vector of probe and/or cell names.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cellmix</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="co">#&gt; class: TapestriExperiment </span></span>
<span><span class="co">#&gt; dim: 5 5 </span></span>
<span><span class="co">#&gt; metadata(7): sample.name pipeline.panel.name ... date.h5.created mean.reads.per.cell.per.probe</span></span>
<span><span class="co">#&gt; assays(3): counts normcounts copyNumber</span></span>
<span><span class="co">#&gt; rownames(5): AMPL158802 AMPL146998 AMPL158817 AMPL158827 AMPL147006</span></span>
<span><span class="co">#&gt; rowData names(10): probe.id chr ... norm.count.sd example.name</span></span>
<span><span class="co">#&gt; colnames(5): AACAACCTAATAGTGGTT-1 AACAACCTAGTCCTAGTT-1 AACAACCTATGGACGAGA-1 AACAACTGGGACATAACG-1</span></span>
<span><span class="co">#&gt;   AACAACTGGGGAACCTAG-1</span></span>
<span><span class="co">#&gt; colData names(3): cell.barcode total.reads cluster</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: CNV</span></span>
<span><span class="co">#&gt; altExpNames(3): alleleFrequency smoothedCopyNumberByChr smoothedCopyNumberByArm</span></span>
<span><span class="co">#&gt; barcodeProbe: not specified</span></span>
<span><span class="co">#&gt; grnaProbe: not specified</span></span>
<span><span class="co">#&gt; gmmParams(2): chr arm</span></span></code></pre></div>
<p>Additional metadata generated automatically when the object is
created can be retrieved using <code>metadata()</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">metadata</span><span class="op">(</span><span class="va">cellmix</span><span class="op">)</span></span>
<span><span class="co">#&gt; $sample.name</span></span>
<span><span class="co">#&gt; [1] "Teresa_s_cell_line_mix"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $pipeline.panel.name</span></span>
<span><span class="co">#&gt; [1] "CO261_NYU_Davoli_03102021_hg19"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $pipeline.version</span></span>
<span><span class="co">#&gt; [1] "2.0.2"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $number.of.cells</span></span>
<span><span class="co">#&gt; [1] "3555"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $number.of.probes</span></span>
<span><span class="co">#&gt; [1] "330"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $date.h5.created</span></span>
<span><span class="co">#&gt; [1] "2021-09-15"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $mean.reads.per.cell.per.probe</span></span>
<span><span class="co">#&gt; [1] "89.22"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="assays">Assays<a class="anchor" aria-label="anchor" href="#assays"></a>
</h2>
<p>The basic unit of the <code>TapestriExperiment</code> is the “assay”
which can be found in the <code>assays</code> slot. Each
<code>assay</code> holds values of the probe x cell matrix. There can be
multiple assays in the slot, each representing some value corresponding
to the same probes and cells. Here we use assays to store raw count
values from sequencing, the values of those counts after normalization,
and the copy number score for each cell and probe unit. Calling
<code>assay()</code> will return the matrix of the first-indexed
<code>assay</code> by default; others can be called by specifying their
name.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># list assays</span></span>
<span><span class="fu">assays</span><span class="op">(</span><span class="va">cellmix</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of length 3</span></span>
<span><span class="co">#&gt; names(3): counts normcounts copyNumber</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get counts assay</span></span>
<span><span class="fu"><a href="../reference/corner.html">corner</a></span><span class="op">(</span><span class="fu">assay</span><span class="op">(</span><span class="va">cellmix</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;            AACAACCTAATAGTGGTT-1 AACAACCTAGTCCTAGTT-1 AACAACCTATGGACGAGA-1 AACAACTGGGACATAACG-1</span></span>
<span><span class="co">#&gt; AMPL158802                   97                  129                   63                   66</span></span>
<span><span class="co">#&gt; AMPL146998                   36                   18                   94                   10</span></span>
<span><span class="co">#&gt; AMPL158817                   71                   17                   62                    1</span></span>
<span><span class="co">#&gt; AMPL158827                  495                  330                  489                  192</span></span>
<span><span class="co">#&gt; AMPL147006                  142                   26                   95                   37</span></span>
<span><span class="co">#&gt;            AACAACTGGGGAACCTAG-1</span></span>
<span><span class="co">#&gt; AMPL158802                   15</span></span>
<span><span class="co">#&gt; AMPL146998                   21</span></span>
<span><span class="co">#&gt; AMPL158817                    5</span></span>
<span><span class="co">#&gt; AMPL158827                  149</span></span>
<span><span class="co">#&gt; AMPL147006                   15</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get copyNumber assay</span></span>
<span><span class="fu"><a href="../reference/corner.html">corner</a></span><span class="op">(</span><span class="fu">assay</span><span class="op">(</span><span class="va">cellmix</span>, <span class="st">"copyNumber"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;            AACAACCTAATAGTGGTT-1 AACAACCTAGTCCTAGTT-1 AACAACCTATGGACGAGA-1 AACAACTGGGACATAACG-1</span></span>
<span><span class="co">#&gt; AMPL158802             1.706083            6.0371189             1.302930            3.8389024</span></span>
<span><span class="co">#&gt; AMPL146998             1.163343            1.5477091             3.571785            1.0686610</span></span>
<span><span class="co">#&gt; AMPL158817             2.765227            1.7617041             2.839333            0.1287974</span></span>
<span><span class="co">#&gt; AMPL158827             1.239042            2.1978910             1.439268            1.5893391</span></span>
<span><span class="co">#&gt; AMPL147006             1.944232            0.9472066             1.529451            1.6753140</span></span>
<span><span class="co">#&gt;            AACAACTGGGGAACCTAG-1</span></span>
<span><span class="co">#&gt; AMPL158802            1.1006006</span></span>
<span><span class="co">#&gt; AMPL146998            2.8309654</span></span>
<span><span class="co">#&gt; AMPL158817            0.8123674</span></span>
<span><span class="co">#&gt; AMPL158827            1.5558830</span></span>
<span><span class="co">#&gt; AMPL147006            0.8567637</span></span></code></pre></div>
<p>Like any matrix, assays can also be subset using bracket
notation.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">assay</span><span class="op">(</span><span class="va">cellmix</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="co">#&gt;            AACAACCTAATAGTGGTT-1 AACAACCTAGTCCTAGTT-1 AACAACCTATGGACGAGA-1 AACAACTGGGACATAACG-1</span></span>
<span><span class="co">#&gt; AMPL158802                   97                  129                   63                   66</span></span>
<span><span class="co">#&gt; AMPL146998                   36                   18                   94                   10</span></span>
<span><span class="co">#&gt; AMPL158817                   71                   17                   62                    1</span></span>
<span><span class="co">#&gt; AMPL158827                  495                  330                  489                  192</span></span>
<span><span class="co">#&gt; AMPL147006                  142                   26                   95                   37</span></span>
<span><span class="co">#&gt;            AACAACTGGGGAACCTAG-1</span></span>
<span><span class="co">#&gt; AMPL158802                   15</span></span>
<span><span class="co">#&gt; AMPL146998                   21</span></span>
<span><span class="co">#&gt; AMPL158817                    5</span></span>
<span><span class="co">#&gt; AMPL158827                  149</span></span>
<span><span class="co">#&gt; AMPL147006                   15</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="alternate-experiments">Alternate Experiments<a class="anchor" aria-label="anchor" href="#alternate-experiments"></a>
</h2>
<p>Alternate Experiments (<code>altExps</code>) allow the use to store
values from the same samples with different features. In other words,
cells in the data have associated read counts for probes and allele
frequencies for variants. The <code>altExp</code> framework allows us to
store both datasets for the same columns/cells, one where the rows are
probes and one where the rows are variants. We use this to store the
smoothed chromosome scores and copy number calls as well, where the
features/rows are chromosomes or chromosome arms. Each
<code>altExp</code> can have its own set of features and several
<code>assay</code>s. The <code>assay</code>s contained within an
<code>altExp</code> must have the same set of features.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># list alternate experiments</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExps</a></span><span class="op">(</span><span class="va">cellmix</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of length 3</span></span>
<span><span class="co">#&gt; names(3): alleleFrequency smoothedCopyNumberByChr smoothedCopyNumberByArm</span></span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># retrieve alternate experiment</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">cellmix</span>, <span class="st">"smoothedCopyNumberByChr"</span><span class="op">)</span></span>
<span><span class="co">#&gt; class: TapestriExperiment </span></span>
<span><span class="co">#&gt; dim: 23 2987 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(3): smoothedCopyNumber discreteCopyNumber gmmCopyNumber</span></span>
<span><span class="co">#&gt; rownames(23): 1 2 ... 22 X</span></span>
<span><span class="co">#&gt; rowData names(0):</span></span>
<span><span class="co">#&gt; colnames(2987): AACAACCTAATAGTGGTT-1 AACAACCTAGTCCTAGTT-1 ... TTGTAATGCTCGTACCTT-1</span></span>
<span><span class="co">#&gt;   TTGTATCACACTTGATCT-1</span></span>
<span><span class="co">#&gt; colData names(0):</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span>
<span><span class="co">#&gt; barcodeProbe: </span></span>
<span><span class="co">#&gt; grnaProbe: </span></span>
<span><span class="co">#&gt; gmmParams(0):</span></span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># list assays in altExp</span></span>
<span><span class="fu">assays</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">cellmix</span>, <span class="st">"smoothedCopyNumberByChr"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of length 3</span></span>
<span><span class="co">#&gt; names(3): smoothedCopyNumber discreteCopyNumber gmmCopyNumber</span></span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># retrieve assay in altExp</span></span>
<span><span class="fu"><a href="../reference/corner.html">corner</a></span><span class="op">(</span><span class="fu">assay</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">cellmix</span>, <span class="st">"smoothedCopyNumberByChr"</span><span class="op">)</span>, <span class="st">"gmmCopyNumber"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;   AACAACCTAATAGTGGTT-1 AACAACCTAGTCCTAGTT-1 AACAACCTATGGACGAGA-1 AACAACTGGGACATAACG-1 AACAACTGGGGAACCTAG-1</span></span>
<span><span class="co">#&gt; 1                    2                    2                    2                    3                    2</span></span>
<span><span class="co">#&gt; 2                    2                    2                    2                    2                    3</span></span>
<span><span class="co">#&gt; 3                    2                    2                    2                    3                    2</span></span>
<span><span class="co">#&gt; 4                    2                    2                    3                    2                    2</span></span>
<span><span class="co">#&gt; 5                    3                    3                    2                    2                    2</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="other-slots">Other Slots<a class="anchor" aria-label="anchor" href="#other-slots"></a>
</h2>
<p>The <code>barcodeProbe</code> and <code>grnaProbe</code> slots are
set automatically by the <code>panel.id</code> parameter of
<code><a href="../reference/CreateTapestriExperiment.html">createTapestriExperiment()</a></code>. They are used as shortcuts for
the <a href="https://joeymays.xyz/karyotapR/articles/barcoding.html" class="external-link">barcoded
read parsing</a> functions of the package. The values for these slots
can be retrieved and set manually as well.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/slotGettersSetters.html">barcodeProbe</a></span><span class="op">(</span><span class="va">cellmix</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "not specified"</span></span>
<span><span class="fu"><a href="../reference/slotGettersSetters.html">grnaProbe</a></span><span class="op">(</span><span class="va">cellmix</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "not specified"</span></span>
<span></span>
<span><span class="co"># setting manually</span></span>
<span><span class="fu"><a href="../reference/slotGettersSetters.html">grnaProbe</a></span><span class="op">(</span><span class="va">cellmix</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"probe_id_123"</span></span></code></pre></div>
<p>The <code>gmmParams</code> slot stores information for the <a href="https://joeymays.xyz/karyotapR/articles/GMMs-for-Copy-Number-Calling.html" class="external-link">GMM
models</a> generated for copy number calling in the form of two nested
matrices one for whole chromosome calls and one for chromosome arm
calls. The data include the Gaussian model parameters fit for each
chromosome or arm, the probability density functions for those models,
the probabilities of each cell belonging to a copy number class, and the
classification results.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/slotGettersSetters.html">gmmParams</a></span><span class="op">(</span><span class="va">cellmix</span><span class="op">)</span></span>
<span><span class="co">#&gt; $chr</span></span>
<span><span class="co">#&gt; # A tibble: 23 × 7</span></span>
<span><span class="co">#&gt;    feature.id smoothed.cn          model            pdf                  model.evidence cn.probability cn.class</span></span>
<span><span class="co">#&gt;    &lt;fct&gt;      &lt;list&gt;               &lt;list&gt;           &lt;list&gt;               &lt;list&gt;         &lt;list&gt;         &lt;list&gt;  </span></span>
<span><span class="co">#&gt;  1 1          &lt;tibble [2,987 × 2]&gt; &lt;tibble [6 × 3]&gt; &lt;tibble [2,987 × 6]&gt; &lt;dbl [2,987]&gt;  &lt;tibble&gt;       &lt;tibble&gt;</span></span>
<span><span class="co">#&gt;  2 2          &lt;tibble [2,987 × 2]&gt; &lt;tibble [6 × 3]&gt; &lt;tibble [2,987 × 6]&gt; &lt;dbl [2,987]&gt;  &lt;tibble&gt;       &lt;tibble&gt;</span></span>
<span><span class="co">#&gt;  3 3          &lt;tibble [2,987 × 2]&gt; &lt;tibble [6 × 3]&gt; &lt;tibble [2,987 × 6]&gt; &lt;dbl [2,987]&gt;  &lt;tibble&gt;       &lt;tibble&gt;</span></span>
<span><span class="co">#&gt;  4 4          &lt;tibble [2,987 × 2]&gt; &lt;tibble [6 × 3]&gt; &lt;tibble [2,987 × 6]&gt; &lt;dbl [2,987]&gt;  &lt;tibble&gt;       &lt;tibble&gt;</span></span>
<span><span class="co">#&gt;  5 5          &lt;tibble [2,987 × 2]&gt; &lt;tibble [6 × 3]&gt; &lt;tibble [2,987 × 6]&gt; &lt;dbl [2,987]&gt;  &lt;tibble&gt;       &lt;tibble&gt;</span></span>
<span><span class="co">#&gt;  6 6          &lt;tibble [2,987 × 2]&gt; &lt;tibble [6 × 3]&gt; &lt;tibble [2,987 × 6]&gt; &lt;dbl [2,987]&gt;  &lt;tibble&gt;       &lt;tibble&gt;</span></span>
<span><span class="co">#&gt;  7 7          &lt;tibble [2,987 × 2]&gt; &lt;tibble [6 × 3]&gt; &lt;tibble [2,987 × 6]&gt; &lt;dbl [2,987]&gt;  &lt;tibble&gt;       &lt;tibble&gt;</span></span>
<span><span class="co">#&gt;  8 8          &lt;tibble [2,987 × 2]&gt; &lt;tibble [6 × 3]&gt; &lt;tibble [2,987 × 6]&gt; &lt;dbl [2,987]&gt;  &lt;tibble&gt;       &lt;tibble&gt;</span></span>
<span><span class="co">#&gt;  9 9          &lt;tibble [2,987 × 2]&gt; &lt;tibble [6 × 3]&gt; &lt;tibble [2,987 × 6]&gt; &lt;dbl [2,987]&gt;  &lt;tibble&gt;       &lt;tibble&gt;</span></span>
<span><span class="co">#&gt; 10 10         &lt;tibble [2,987 × 2]&gt; &lt;tibble [6 × 3]&gt; &lt;tibble [2,987 × 6]&gt; &lt;dbl [2,987]&gt;  &lt;tibble&gt;       &lt;tibble&gt;</span></span>
<span><span class="co">#&gt; # ℹ 13 more rows</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $arm</span></span>
<span><span class="co">#&gt; # A tibble: 41 × 7</span></span>
<span><span class="co">#&gt;    feature.id smoothed.cn          model            pdf                  model.evidence cn.probability cn.class</span></span>
<span><span class="co">#&gt;    &lt;fct&gt;      &lt;list&gt;               &lt;list&gt;           &lt;list&gt;               &lt;list&gt;         &lt;list&gt;         &lt;list&gt;  </span></span>
<span><span class="co">#&gt;  1 chr1p      &lt;tibble [2,987 × 2]&gt; &lt;tibble [6 × 3]&gt; &lt;tibble [2,987 × 6]&gt; &lt;dbl [2,987]&gt;  &lt;tibble&gt;       &lt;tibble&gt;</span></span>
<span><span class="co">#&gt;  2 chr1q      &lt;tibble [2,987 × 2]&gt; &lt;tibble [6 × 3]&gt; &lt;tibble [2,987 × 6]&gt; &lt;dbl [2,987]&gt;  &lt;tibble&gt;       &lt;tibble&gt;</span></span>
<span><span class="co">#&gt;  3 chr2p      &lt;tibble [2,987 × 2]&gt; &lt;tibble [6 × 3]&gt; &lt;tibble [2,987 × 6]&gt; &lt;dbl [2,987]&gt;  &lt;tibble&gt;       &lt;tibble&gt;</span></span>
<span><span class="co">#&gt;  4 chr2q      &lt;tibble [2,987 × 2]&gt; &lt;tibble [6 × 3]&gt; &lt;tibble [2,987 × 6]&gt; &lt;dbl [2,987]&gt;  &lt;tibble&gt;       &lt;tibble&gt;</span></span>
<span><span class="co">#&gt;  5 chr3p      &lt;tibble [2,987 × 2]&gt; &lt;tibble [6 × 3]&gt; &lt;tibble [2,987 × 6]&gt; &lt;dbl [2,987]&gt;  &lt;tibble&gt;       &lt;tibble&gt;</span></span>
<span><span class="co">#&gt;  6 chr3q      &lt;tibble [2,987 × 2]&gt; &lt;tibble [6 × 3]&gt; &lt;tibble [2,987 × 6]&gt; &lt;dbl [2,987]&gt;  &lt;tibble&gt;       &lt;tibble&gt;</span></span>
<span><span class="co">#&gt;  7 chr4p      &lt;tibble [2,987 × 2]&gt; &lt;tibble [6 × 3]&gt; &lt;tibble [2,987 × 6]&gt; &lt;dbl [2,987]&gt;  &lt;tibble&gt;       &lt;tibble&gt;</span></span>
<span><span class="co">#&gt;  8 chr4q      &lt;tibble [2,987 × 2]&gt; &lt;tibble [6 × 3]&gt; &lt;tibble [2,987 × 6]&gt; &lt;dbl [2,987]&gt;  &lt;tibble&gt;       &lt;tibble&gt;</span></span>
<span><span class="co">#&gt;  9 chr5p      &lt;tibble [2,987 × 2]&gt; &lt;tibble [6 × 3]&gt; &lt;tibble [2,987 × 6]&gt; &lt;dbl [2,987]&gt;  &lt;tibble&gt;       &lt;tibble&gt;</span></span>
<span><span class="co">#&gt; 10 chr5q      &lt;tibble [2,987 × 2]&gt; &lt;tibble [6 × 3]&gt; &lt;tibble [2,987 × 6]&gt; &lt;dbl [2,987]&gt;  &lt;tibble&gt;       &lt;tibble&gt;</span></span>
<span><span class="co">#&gt; # ℹ 31 more rows</span></span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sessioninfo</span><span class="fu">::</span><span class="fu"><a href="https://r-lib.github.io/sessioninfo/reference/session_info.html" class="external-link">session_info</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; ─ Session info ────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt;  setting  value</span></span>
<span><span class="co">#&gt;  version  R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">#&gt;  os       macOS Ventura 13.5.2</span></span>
<span><span class="co">#&gt;  system   x86_64, darwin20</span></span>
<span><span class="co">#&gt;  ui       RStudio</span></span>
<span><span class="co">#&gt;  language (EN)</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       America/New_York</span></span>
<span><span class="co">#&gt;  date     2023-09-18</span></span>
<span><span class="co">#&gt;  rstudio  2023.06.0+421 Mountain Hydrangea (desktop)</span></span>
<span><span class="co">#&gt;  pandoc   3.1.1 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ─ Packages ────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt;  package              * version   date (UTC) lib source</span></span>
<span><span class="co">#&gt;  abind                  1.4-5     2016-07-21 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  Biobase              * 2.60.0    2023-05-11 [1] Bioconductor</span></span>
<span><span class="co">#&gt;  BiocGenerics         * 0.46.0    2023-05-11 [1] Bioconductor</span></span>
<span><span class="co">#&gt;  bitops                 1.0-7     2021-04-24 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  cachem                 1.0.8     2023-05-01 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  Cairo                  1.6-0     2022-07-05 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  callr                  3.7.3     2022-11-02 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  circlize               0.4.15    2022-05-10 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  cli                    3.6.1     2023-03-23 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  clue                   0.3-64    2023-01-31 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  cluster                2.1.4     2022-08-22 [1] CRAN (R 4.3.1)</span></span>
<span><span class="co">#&gt;  codetools              0.2-19    2023-02-01 [1] CRAN (R 4.3.1)</span></span>
<span><span class="co">#&gt;  colorspace             2.1-0     2023-01-23 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  commonmark             1.9.0     2023-03-17 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  ComplexHeatmap         2.16.0    2023-05-11 [1] Bioconductor</span></span>
<span><span class="co">#&gt;  cowplot                1.1.1     2020-12-30 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  crayon                 1.5.2     2022-09-29 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  DelayedArray           0.26.7    2023-07-28 [1] Bioconductor</span></span>
<span><span class="co">#&gt;  desc                   1.4.2     2022-09-08 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  devtools             * 2.4.5     2022-10-11 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  digest                 0.6.33    2023-07-07 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  distributional         0.3.2     2023-03-22 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  doParallel             1.0.17    2022-02-07 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  dplyr                * 1.1.3     2023-09-03 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  ellipsis               0.3.2     2021-04-29 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  evaluate               0.21      2023-05-05 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  fansi                  1.0.4     2023-01-22 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  farver                 2.1.1     2022-07-06 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  fastmap                1.1.1     2023-02-24 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  fitdistrplus         * 1.1-11    2023-04-25 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  forcats              * 1.0.0     2023-01-29 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  foreach                1.5.2     2022-02-02 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  formatR                1.14      2023-01-17 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  fs                     1.6.3     2023-07-20 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  generics               0.1.3     2022-07-05 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  GenomeInfoDb         * 1.36.2    2023-08-25 [1] Bioconductor</span></span>
<span><span class="co">#&gt;  GenomeInfoDbData       1.2.10    2023-07-09 [1] Bioconductor</span></span>
<span><span class="co">#&gt;  GenomicRanges        * 1.52.0    2023-05-11 [1] Bioconductor</span></span>
<span><span class="co">#&gt;  GetoptLong             1.0.5     2020-12-15 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  ggdist               * 3.3.0     2023-05-13 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  ggplot2              * 3.4.3     2023-08-14 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  GlobalOptions          0.1.2     2020-06-10 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  glue                   1.6.2     2022-02-24 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  gtable                 0.3.4     2023-08-21 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  highr                  0.10      2022-12-22 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  htmltools              0.5.5     2023-03-23 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  htmlwidgets            1.6.2     2023-03-17 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  httpuv                 1.6.11    2023-05-11 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  hunspell               3.0.2     2022-09-04 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  IRanges              * 2.34.1    2023-06-22 [1] Bioconductor</span></span>
<span><span class="co">#&gt;  iterators              1.0.14    2022-02-05 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  karyotapR            * 1.0.1     2023-09-07 [1] CRAN (R 4.3.1)</span></span>
<span><span class="co">#&gt;  knitr                  1.43      2023-05-25 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  labeling               0.4.3     2023-08-29 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  later                  1.3.1     2023-05-02 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  lattice                0.21-8    2023-04-05 [1] CRAN (R 4.3.1)</span></span>
<span><span class="co">#&gt;  lifecycle              1.0.3     2022-10-07 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  magick                 2.7.4     2023-03-09 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  magrittr               2.0.3     2022-03-30 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  MASS                 * 7.3-60    2023-05-04 [1] CRAN (R 4.3.1)</span></span>
<span><span class="co">#&gt;  Matrix                 1.6-0     2023-07-08 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  MatrixGenerics       * 1.12.3    2023-07-31 [1] Bioconductor</span></span>
<span><span class="co">#&gt;  matrixStats          * 1.0.0     2023-06-02 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  memoise                2.0.1     2021-11-26 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  mime                   0.12      2021-09-28 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  miniUI                 0.1.1.1   2018-05-18 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  munsell                0.5.0     2018-06-12 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  pillar                 1.9.0     2023-03-22 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  pkgbuild               1.4.2     2023-06-26 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  pkgconfig              2.0.3     2019-09-22 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  pkgload                1.3.2.1   2023-07-08 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  png                    0.1-8     2022-11-29 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  prettyunits            1.1.1     2020-01-24 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  processx               3.8.2     2023-06-30 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  profvis                0.3.8     2023-05-02 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  promises               1.2.0.1   2021-02-11 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  ps                     1.7.5     2023-04-18 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  purrr                * 1.0.2     2023-08-10 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  R6                     2.5.1     2021-08-19 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  RColorBrewer           1.1-3     2022-04-03 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  Rcpp                   1.0.11    2023-07-06 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  RCurl                  1.98-1.12 2023-03-27 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  remotes                2.4.2.1   2023-07-18 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  rjson                  0.2.21    2022-01-09 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  rlang                  1.1.1     2023-04-28 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  rmarkdown              2.23      2023-07-01 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  rprojroot              2.0.3     2022-04-02 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  rstudioapi             0.15.0    2023-07-07 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  S4Arrays               1.0.6     2023-08-30 [1] Bioconductor</span></span>
<span><span class="co">#&gt;  S4Vectors            * 0.38.1    2023-05-11 [1] Bioconductor</span></span>
<span><span class="co">#&gt;  scales                 1.2.1     2022-08-20 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  sessioninfo            1.2.2     2021-12-06 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  shape                  1.4.6     2021-05-19 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  shiny                  1.7.4.1   2023-07-06 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  SingleCellExperiment * 1.22.0    2023-05-11 [1] Bioconductor</span></span>
<span><span class="co">#&gt;  spelling               2.2.1     2023-03-22 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  stringi                1.7.12    2023-01-11 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  stringr                1.5.0     2022-12-02 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  SummarizedExperiment * 1.30.2    2023-06-06 [1] Bioconductor</span></span>
<span><span class="co">#&gt;  survival             * 3.5-5     2023-03-12 [1] CRAN (R 4.3.1)</span></span>
<span><span class="co">#&gt;  tibble                 3.2.1     2023-03-20 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  tidyr                  1.3.0     2023-01-24 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  tidyselect             1.2.0     2022-10-10 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  urlchecker             1.0.1     2021-11-30 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  usethis              * 2.2.2     2023-07-06 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  utf8                   1.2.3     2023-01-31 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  vctrs                  0.6.3     2023-06-14 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  viridisLite            0.4.2     2023-05-02 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  whisker                0.4.1     2022-12-05 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  withr                  2.5.0     2022-03-03 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  xfun                   0.39      2023-04-20 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  xml2                   1.3.5     2023-07-06 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  xtable                 1.8-4     2019-04-21 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  XVector                0.40.0    2023-05-11 [1] Bioconductor</span></span>
<span><span class="co">#&gt;  yaml                   2.3.7     2023-01-23 [1] CRAN (R 4.3.0)</span></span>
<span><span class="co">#&gt;  zlibbioc               1.46.0    2023-05-11 [1] Bioconductor</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  [1] /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ───────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Joseph Mays.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
